src/classes/STG_InstallScript.cls

```diff
@@ -420,6 +437,35 @@ global without sharing class STG_InstallScript implements InstallHandler {
         }
     }
 
+    /*******************************************************************************************************
+    * @description Creates a new set of Migrated Custom Field Mappings based on if advanced mapping is turned on
+    */
+    @TestVisible
+    private void createMigratedCustomFieldMappings() {
```

Comment:
@ptewson-sfdo The Utility class called here checks for whether or not the corresponding Migrated Custom Field Mappings have been created, and if they have, then we will not do anything. I will add to our AC to see if it's possible to test and simulate multiple installs (cc @Julian88Tex).

As for making this dynamic, what do you think @callen-sfdo? Would it be possible to come back to this after GA?

Reply:
@mldyang by simulating multiple installs, do you mean with a test? Simulating an upgrade is definitely possible using cumulusci post merge to master. Unfortunately, we don’t have any way to test pre-merge so there’s a chance an issue will be discovered and a new PR will need to be created post-merge.

Reply:
Yeah, this is something that we have had some debate about.  I was concerned about just running against all existing Data Import Field Mappings in the default field mapping set because they may have altered the existing mappings and we wouldn't want to duplicate them.  Ideally we would only run this script with this release of NPSP (ie the one that creates the new fields) and future package versions would not run this script.  That said, I guess there is also risk that if someone upgrades from a prior version to a future version and skips this version that the script would never run.

Reply:
Yeah - from my perspective I think it's worth considering a more dynamic option, but I definitely see the utility to keeping this focused to a limited set of metadata. Thanks for walking through the thought behind this and I'm good to go with this approach!

Reply:
@ptewson-sfdo and @callen-sfdo Could we talk through this a bit more? I'm a little concerned about only having this in one release. There are definitely customers who don't receive a release/ skip a release for a variety of reasons -- ranging from gacks related to something in their own org to maybe a timeout issue on their Instance. Release Engineering typically does a re-push to failed customer orgs, but there are still definitely customers who won't get the 3.175 release for one reason or another. If that happens and if this is only code running in the install script of one release, what should Support and CCE do to help any orgs who don't get these mappings created? I think this update only affects existing AM customers, but still... it can happen. Thanks!

Reply:
@jesslopez-sf Great questions. I think (and @callen-sfdo can correct me if I'm wrong) this delivery, as it's currently written, will actually catch customers up on these field mappings in this and future releases - it's static in terms of the mappings it will upgrade, but currently will do the update on every release.

Reply:
Hi @jesslopez-sf yup the intent is that this will run on every upgrade. 

Currently, if the user has advanced mapping turned on already, they will need to disable and re-enable Advanced Mapping toggle on the NPSP Setting page to trigger the DI field migrations. The purpose of this post-install script is to avoid this manual step. 

The logic on how we're doing this is:
- Upon upgrade, the post install script here will check to see if Advanced Mapping is on
- If Adavnced Mapping is on, we will take a static list of fields (ex Payment_Elevate_ID) to see if there are corresponding Migrated Custom Field Mapping records in the system already.
- If so, do nothing; if not, we will create the Migrated Custom Field Mapping records.